src/runtime/runtime.cs

```diff
@@ -1026,44 +1030,69 @@ internal static bool PyObject_IsIterable(IntPtr pointer)
             return tp_iter != IntPtr.Zero;
         }
 
-        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
-        internal static extern int PyObject_HasAttrString(IntPtr pointer, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(Utf8Marshaler))] string name);
+        
+        internal static int PyObject_HasAttrString(BorrowedReference pointer, string name)
+        {
+            using var namePtr = new StrPtr(name, Encoding.UTF8);
+            return Delegates.PyObject_HasAttrString(pointer, namePtr);
+        }
 
-        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
-        internal static extern IntPtr PyObject_GetAttrString(IntPtr pointer, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(Utf8Marshaler))] string name);
+        internal static IntPtr PyObject_GetAttrString(IntPtr pointer, string name)
+        {
+            IntPtr nameMem = Marshal.StringToHGlobalAnsi(name);
```

Comment:
This will not work for unicode strings, is it just an oversight that you don't use your `StrPtr` here as for other functions?

Reply:
This was a mistake, probably leftover from the first try.